### DISCUSSION TITLE
Cortex M55/adiv6 support

### DISCUSSION THREAD

**Discussion by florg-32** [blackmagic-debug/blackmagic#1969] [Q&A] [answered]
Hey guys,

I have a Cortex M55 chip I'd like to debug with my BMP. Should this be supported already (on nightly builds)? 

My gdb `monitor jtag_scan` does not find the device, voltage is detected correctly though and resetting works. The debug app outputs the following, which I don't really understand; shouldn't it be ADIv6?

```
❯ blackmagic -tjv8
Black Magic Debug App d896778
 for Black Magic Probe, ST-Link v2 and v3, CMSIS-DAP, J-Link and FTDI (MPSSE)
Using 1d50:6018 81D749A3 Black Magic Debug
 Black Magic Probe d896778
Running in Test Mode
Target voltage: 1.8V
Speed set to 1.951MHz for JTAG
Resetting TAP
Change state to Shift-DR
Scanning out ID codes
Return to Run-Test/Idle
Change state to Shift-IR
Scanning out IRs
Return to Run-Test/Idle
Change state to Shift-DR
Return to Run-Test/Idle
ID code 0x4ba06477: ADIv5 JTAG-DP port.
Abort: 00000004
Failed to read DPIDR
Given target number 1 not available max 0
```
Thanks,
Florian

**Comment by dragonmux** [comment, marked as answer]
👋🏼  Cortex-M55 should indeed be supported in some capacity at least. What you're seeing in the output there is that it's finding a JTAG DP, however we never got to see an example of what a ADIv6 JTAG-DPv2's ID code looks like before so BMD is likely not realising that's an ADIv6 DP. The DPIDR read-out failure isn't helping either as that's indicative of an issue talking with ADIv6's JTAG-DPv2's.

Essentially what we're saying is that SWD-DPv3 has been proven out and tested, but because our test target was SWD-only (RP2350), we had to make some guesses about JTAG-DPv2 correctness/functionality and it looks like there's some more work needed there to get JTAG going. If you target has a SWJ-DP (ie one that can talk either JTAG or SWD) then for now, please scan using SWD. We'll transfer this to an issue to address the JTAG side of things being unhappy and needing some work to get woring.

**Reply by florg-32** [reply]
Thank you for the super fast response!
